src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/DataTableCaption.test.tsx

```diff
@@ -0,0 +1,378 @@
+import React from 'react';
+import {
+  CategoryFilter,
+  Indicator,
+  LocationFilter,
+  TimePeriodFilter,
+} from '@common/modules/table-tool/types/filters';
+import { FullTableMeta } from '@common/modules/table-tool/types/fullTable';
+import { render, screen, fireEvent } from '@testing-library/react';
+import DataTableCaption from '../DataTableCaption';
+
+describe('DataTableCaption', () => {
+  const generateTableTitleBasicInput: FullTableMeta = {
+    geoJsonAvailable: false,
+    publicationName: 'Pupil absence in schools in England',
+    subjectName: 'Absence by characteristic',
+    footnotes: [],
+    boundaryLevels: [],
+    filters: {
+      Characteristic: {
+        name: 'characteristic',
+        options: [
+          new CategoryFilter({
+            value: 'total',
+            label: 'Total',
+            group: 'Gender',
+            category: 'Characteristic',
+          }),
+        ],
+      },
+    },
+    indicators: [
+      new Indicator({
+        label: 'Authorised absence rate',
+        value: 'authAbsRate',
+        unit: '%',
+        name: 'sess_authorised_percent',
+      }),
+    ],
+    locations: [
+      new LocationFilter({
+        value: 'E92000001',
+        label: 'England',
+        level: 'country',
+      }),
+    ],
+    timePeriodRange: [
+      new TimePeriodFilter({
+        code: 'AY',
+        year: 2015,
+        label: '2015/16',
+        order: 0,
+      }),
+    ],
+  };
+
+  test('with given title', () => {
+    const { container } = render(
+      <DataTableCaption
+        id="DataTableCaption"
+        title="Test title"
+        {...generateTableTitleBasicInput}
+      />,
+    );
+
+    expect(screen.queryByText('Test title')).not.toBeNull();
+    expect(container).not.toHaveTextContent('full table title');
```

Comment:
Nope, no need to use the container. You should use the `ByRole` selector instead:

```javascript
screen.queryByRole('button', { name: /full table title/ })
```